tests/Regressions.hs

```diff
@@ -71,6 +76,48 @@ propEqAfterDelete (Keys keys) =
 mapFromKeys :: [Int] -> HM.HashMap Int ()
 mapFromKeys keys = HM.fromList (zip keys (repeat ()))
 
+------------------------------------------------------------------------
+-- Issue #254
+
+data KC = KC Int
+  deriving (Eq, Ord, Show)
+instance Hashable KC where
+  hashWithSalt salt _ = salt
```

Comment:
The laziness in the `KC` value is completely unimportant, right? The only relevant space leak is  the one in `updateSnocWithKey`?!

Reply:
Yes.